src/sentences/aam.rs

```diff
@@ -0,0 +1,130 @@
+use arrayvec::ArrayString;
+use nom::{
+    bytes::complete::is_not,
+    character::complete::{anychar, char},
+    combinator::opt,
+    number::complete::float,
+};
+
+use crate::{parse::NmeaSentence, sentences::utils::array_string, Error, SentenceType};
+
+const MAX_LEN: usize = 64;
+
+/// AAM - Waypoint Arrival Alarm
+///
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_aam_waypoint_arrival_alarm>
+///
+/// ```text
+///        1 2 3   4 5    6
+///        | | |   | |    |
+/// $--AAM,A,A,x.x,N,c--c*hh<CR><LF>
+/// ```
+#[derive(Debug, PartialEq)]
+pub struct AamData {
+    pub arrival_circle_entered: Option<bool>,
+    pub perpendicular_passed: Option<bool>,
+    pub arrival_circle_radius: Option<f32>,
+    pub radius_units: Option<char>,
+    pub waypoint_id: Option<ArrayString<MAX_LEN>>,
+}
+
+/// Parse AAM message
+pub fn parse_aam(sentence: NmeaSentence) -> Result<AamData, Error> {
+    if sentence.message_id != SentenceType::AAM {
+        Err(Error::WrongSentenceHeader {
+            expected: SentenceType::AAM,
+            found: sentence.message_id,
+        })
+    } else {
+        Ok(do_parse_aam(sentence.data)?)
+    }
+}
+
+fn do_parse_aam(i: &str) -> Result<AamData, Error> {
+    let (i, field1) = anychar(i)?;
+    let arrival_circle_entered = if field1 == 'A' {
+        Some(true)
+    } else {
+        Some(false)
+    };
+    let (i, _) = char(',')(i)?;
+
+    let (i, field2) = anychar(i)?;
+    let perpendicular_passed = if field2 == 'A' {
```

Comment:
We would like to check for both A & V, because anything else should be an error (e.g. if someone passes `G`)

You can check this example:
https://github.com/AeroRust/nmea/blob/main/src/sentences/gll.rs#L71-L76

Reply:
Added an `unreachable` for values other than A or V.